exercises/simple-linked-list/example.scala

```diff
@@ -0,0 +1,94 @@
+import scala.reflect.ClassTag
+
+class Node[+T]
+case object Empty extends Node[Nothing]
+case class NonEmpty[T](value: T, var next: Node[T]) extends Node[T]
```

Comment:
And yet again for my clean conscience: Use an immutable ADT and no side effects anywhere?

And again, you can ignore this comment. :)